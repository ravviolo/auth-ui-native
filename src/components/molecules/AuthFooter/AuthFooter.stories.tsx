import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { StoryScreen } from 'storybook/StoryScreen';

import { AuthFooter } from './AuthFooter';

storiesOf('AuthFooter', module)
  .addDecorator((getStory) => <StoryScreen center>{getStory()}</StoryScreen>)
  .add('Default', () => (
    <AuthFooter
      btnText={text('btnText', 'Click me!')}
      testID={text('testID', 'auth-footer-test-id')}
      text={text('text', 'This is footer with a button.')}
      onPress={action('onPressFooterBtn')}
    />
  ));
