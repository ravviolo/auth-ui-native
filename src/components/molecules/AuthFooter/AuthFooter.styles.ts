import { StyleSheet } from 'react-native';
import { colors } from 'theme';

export const styles = StyleSheet.create({
  footerBtn: {
    textDecorationLine: 'underline',
    textTransform: 'uppercase',
  },
  footerChild: {
    color: colors.textLight,
    marginHorizontal: 2,
  },
  footerContainer: {
    justifyContent: 'center',
  },
});
