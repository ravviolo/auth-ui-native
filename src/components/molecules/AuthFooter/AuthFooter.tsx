import { Flex, TextButton } from 'components/atoms';
import { StyleSheet, Text } from 'react-native';
import { colors } from 'theme';

interface Props {
  text: string;
  btnText: string;
  testID: string;
  onPress: () => void;
}

export const AuthFooter = ({ btnText, testID, onPress, text }: Props) => {
  return (
    <Flex direction="row" style={styles.footerContainer} testID={testID}>
      <Text style={styles.footerChild}>{text}</Text>
      <TextButton
        style={[styles.footerChild, styles.footerBtn]}
        testID={testID}
        title={btnText}
        onPress={onPress}
      />
    </Flex>
  );
};

const styles = StyleSheet.create({
  footerBtn: {
    textDecorationLine: 'underline',
    textTransform: 'uppercase',
  },
  footerChild: {
    color: colors.textLight,
    marginHorizontal: 2,
  },
  footerContainer: {
    justifyContent: 'center',
  },
});
