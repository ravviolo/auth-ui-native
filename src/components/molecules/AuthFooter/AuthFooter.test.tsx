import { render, screen, fireEvent, within } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { AuthFooter } from './AuthFooter';

import type { Props } from './AuthFooter';

describe('AuthFooter', () => {
  const mockOnPress = jest.fn<Props['onPress'], []>();

  const props: Props = {
    btnText: 'Click me!',
    text: 'I am a footer.',
    testID: 'auth-footer-test-id',
    onPress: mockOnPress,
  };

  it('should render footer containing text and button', () => {
    render(<AuthFooter {...props} />);

    const authFooter = screen.getByTestId('auth-footer-test-id');
    expect(authFooter).toHaveTextContent('I am a footer.');

    const footerBtn = within(authFooter).getByTestId('btn-footer-test-id');
    expect(footerBtn).toHaveTextContent('Click me!');
  });

  it('should handle press events, run handler once', () => {
    render(<AuthFooter {...props} />);

    const authFooter = screen.getByTestId('auth-footer-test-id');
    const footerBtn = within(authFooter).getByTestId('btn-footer-test-id');

    fireEvent.press(footerBtn);

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    const tree = testRenderer.create(<AuthFooter {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
