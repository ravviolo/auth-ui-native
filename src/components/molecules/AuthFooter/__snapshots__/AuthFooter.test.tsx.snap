// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthFooter should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "justifyContent": "center",
      },
    ]
  }
  testID="auth-footer-test-id"
>
  <Text
    style={
      Object {
        "color": "#9E9E9E",
        "marginHorizontal": 2,
      }
    }
  >
    I am a footer.
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={20}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={false}
    testID="btn-footer-test-id"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#9E9E9E",
            "marginHorizontal": 2,
          },
          Object {
            "textDecorationLine": "underline",
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Click me!
    </Text>
  </View>
</View>
`;
